### ISSUE TITLE
How to speed up training

### ISSUE THREAD

**Issue by Haoqing-Wang** [loeweX/Greedy_InfoMax#22]
Nice work! I wonder how to speed up training and reduce memory usage. As I can see, the released code uses `.detach()` to prevant backpropagation, but I didn't find it can speed up training or reduce memory usage. Are there any other operations? Looking forward for your reply.

**Comment by loeweX** [comment]
One way to reduce memory usage could be to train the modules individually, i.e. one after another. While the overall procedure for this is not implemented in this codebase, you could start by setting `--train_module 0` to train the first module. After that, it would be advisable to save the outputs of this model as a new dataset and train the next module on it. 

If you have several GPUs available, you could potentially parallelize this process as well by training one module per GPU and by saving the outputs of each module as the inputs for the next module in regular intervals.

Here's a rough schematic of what I mean, where each of the gray boxes would represent a separate GPU:
![image](https://user-images.githubusercontent.com/16319903/187213137-60995786-5ab2-4130-966d-32a7777c6cc7.png)

Hope this helps!
